lib/phoenix_test/case.ex

```diff
@@ -8,42 +8,27 @@ defmodule PhoenixTest.Case do
 
   alias PhoenixTest.Case
   alias PhoenixTest.Playwright
+  alias PhoenixTest.Playwright.Config
 
-  using opts do
+  using _opts do
     quote do
       import PhoenixTest
       import PhoenixTest.Case
       import PhoenixTest.Playwright, only: [screenshot: 2, screenshot: 3]
-
-      setup do
-        [phoenix_test: unquote(opts)]
-      end
     end
   end
 
-  @playwright_opts [
-    browser: :chromium,
-    headless: true,
-    slow_mo: 0
-  ]
-
   setup_all context do
-    global_config = Application.fetch_env!(:phoenix_test, :playwright)
-    global_browser_config = List.wrap(global_config[:browser])
-    trace = Map.get(context, :trace, global_config[:trace])
+    config = context |> Map.take(Config.keys()) |> Config.parse()
 
     case context do
       %{playwright: true} ->
-        opts = Keyword.merge(@playwright_opts, global_browser_config)
-        [browser_id: Case.Playwright.launch_browser(opts), trace: trace]
+        browser_opts = Keyword.take(config, ~w(browser headless slow_mo)a)
+        browser_id = Case.Playwright.launch_browser(browser_opts)
+        Keyword.put(config, :browser_id, browser_id)
 
-      %{playwright: context_browser_config} when is_list(context_browser_config) ->
-        opts =
-          @playwright_opts
-          |> Keyword.merge(global_browser_config)
-          |> Keyword.merge(context_browser_config)
-
-        [browser_id: Case.Playwright.launch_browser(opts), trace: trace]
+      %{playwright: _} ->
+        raise ArgumentError, "Pass any playwright options as top-level tags, e.g. `@moduletag browser: :firefox`"
```

Comment:
Am I reading this correctly that you can no longer modify the config on a per-test level (e.g., `@tag playwright: [headless: true]`)?

Reply:
Sorry, yes you can. It's just hoisted.
Pass `@tag headless: true` instead of `@tag playwright: [headless: true]`.

This allows you to override individual settings without.

E.g. in the old syntax
```ex
@moduletag playwright: [slow_mo: 1000]
@tag playwright: [headless: false]
```

The slowMo config would have been lost, because ExUnit doesn't merge tag values.

In the new snytax
```ex
@moduletag slow_mo: 1000
@tag headless: false
```

it is preserved.

Reply:
I tripped myself up in my own explanation:

You could not, and can not, override browser config via `@describetag` or `@tag`.
Only via `@moduletag`, because the browser is launched in `setup_all`, once per test module.

So there remains only a very limited set of options you can config values you can override via `@describetag` and `@tag`:
- ~browser~
- ~cli~
- ~headless~
- ~js_jogger~
- screenshot
- ~screenshot_dir~
- ~timeout~
- ~slow_mo~
- trace
- ~trace_dir~

Only `trace` and `screenshot`.

Have to document this better, and ideally warn if trying to override where not supported.

Reply:
To make this more explicit it might be better to not configure browser options via `@moduletag`, but rather via `use PhoenixTest.Case, browser: :chromium` `using` opts.

The downside is that it then is not possible to use ExUnit 1.18 parameterized tests to run the module concurrently in different browsers.

Reply:
Ahh, interesting... yeah, that makes sense, and explains why this has occasionally tripped me up.